packages/web-service/src/pages/file-upload/pre-scan/file-upload.js

```diff
@@ -0,0 +1,133 @@
+import Joi from 'joi'
+import fs from 'fs'
+import path from 'path'
+import handler, { errorShim } from '../../../handlers/page-handler.js'
+import { CHECK_YOUR_ANSWERS, FILE_UPLOAD } from '../../../uris.js'
+import { scanFile } from '../../../services/virus-scan.js'
+import { MAX_FILE_UPLOAD_SIZE_MB, TIMEOUT_MS } from '../../../constants.js'
+
+export const setData = async request => {
+  const currentFilePlusDirectory = path.join(process.env.SCANDIR, path.basename(request.payload['scan-file'].path))
+
+  // We need to take a filename like: hello.txt
+  // And rename it to something like: {unixTimestamp}.{originalFileName}
+  // So if the user uploads it to us, but doesn't submit it fully all the way to s3 - we can check the timestamp and delete it
+  // It also stops collisions if multiple users all upload files with the same name
+  const newFilename = `${+new Date()}.${request.payload['scan-file'].filename}`
+  const newFilenamePlusDirectory = path.join(process.env.SCANDIR, newFilename)
+
+  fs.renameSync(currentFilePlusDirectory, newFilenamePlusDirectory, err => {
+    if (err) {
+      console.err(err)
+    }

```

Comment:
Should we throw an exception here? This is a server error if we can't rename the file so we probably want to throw a Boom 500 to land on the server error page. See the api-requests package ...
`
console.error('detailed error message', error)
        Boom.boomify(error, { statusCode: 500 })
        throw error
`
etc